### ISSUE TITLE
server queue

### ISSUE THREAD

**Issue by marcporro** [Hilicot/McSplit#6]
### GNN

- [x] train w/ heuristic
- [x] test v
- [x] test w
- [x] test v+w
- [x] train w/o heuristic
- [x] test v
- [x] test w
- [x] test v+w
- [x] train w/ synth
- [x] test v
- [x] test w
- [x] test v+w
- [x] run Python McSplit alone (baseline)
- [x] run Python random (baseline)
- [x] train+test GNN con parametri alti (dimensione 64 dei layer)
- [x] modifico codice del train => calcolo embedding dei due bidomini, traino con loss = embedding1-embedding2. Uso pooling layer
- [ ] ~~run with max_iter due to gnn being slow~~
- [x] runno test usando GNN come euristica statica

### Parallel

- [x] test parallel w/ different block size
- [x] test parallel w/ different thread count (also 1, to compare with recursive) -> still missing, 64T 32B
- [x] test parallel w/ different thread heuristics (manca katz)
- [x] test parallel w/ different timeouts (done with BR, do it with other heuristics)
- [x] test parallel until_pruning
- [x] test original parallel mccreesh with different heuristics 
- [x] test original parallel mccreesh+PR with different thread count
- [ ] compare our parallel, original parallel, sequential with 5 minutes timeout
- [ ] (opt) test weaker until_pruning?
- [x] run our parallel and original parallel on small dataset
- [x] run our parallel without DAL (to compare to the original parallel)
- [x] parallel BS w/ separated rewards
- [x] original PR w/ 64 threads
- [x] pruning on range of B (another Block size might be the optimal)

### Needed for analysis

- [x] MB 32T katz finetuning
- [x] BS 32T all block sizes (and then all tests on 16B if needed)
- [x] BS w/ 1024 B
- [x] BS noDAL sulle euristiche mancanti
- [x] original trimble seq mcsplit on glsearch timeout
- [ ] MB tutte euristiche sui big
- [ ] mcsplitGNN e static sui big
- [ ] original degree (no dal) e LL su big

### Others

- [ ] ~~test mcsplit w/ dynamic heuristic (small dataset)~~
- [ ] test mcsplitX+PR/BS/CS with different timeouts
- [ ] ~~extend original parallel mccreesh with DAL~~
- [ ] ~~faccio stampare da our parallel i thread work times (per load balancing graph)~~
- [ ] test sui big full (non solo finetuning) dei best, incluso mcsplit degree originale, McSplitLL, McSplitDAL... -> fatto PR per ora
